piksi_multi_cpp/src/receiver/receiver_ros.cc

```diff
@@ -29,9 +29,9 @@ ReceiverRos::ReceiverRos(const ros::NodeHandle& nh, const Device::Ptr& device)
 
   if (1) {
     auto logger = std::make_shared<FileObservationLogger>();
-    /*ROS_WARN_STREAM(logger->open("/tmp/tempfile.sbp"));
-    obs_cbs_->addObservationCallbackListener(
-        CBtoRawObsConverter::createFor(logger));*/
+    ROS_WARN_STREAM(logger->open("/tmp/tempfile.sbp"));
+    obs_cbs_->addMsgCallbackListener(
+        CBtoRawObsConverter::createFor(logger, uint16_t(0x42))); // Hardcoded sender Id for now, as it probably does not matter
```

Comment:
You can get the sender_id from the firmware settings. https://github.com/ethz-asl/ethz_piksi_ros/blob/b132ff25a65ad0e6def9f952882adf6b1b115e50/piksi_multi_cpp/src/receiver/receiver_base_station.cc#L25

Reply:
Thanks for the hint! I moved the sender id down to the ReceiverRos. Now all receivers store their sender_id.